### PR TITLE
Documentation and miscellaneous

### PR DESCRIPTION
**Pull request by ilyaZar**
Hi I am sorry about being a bit pushy with updating the other files (and whitespaces); we can always ignore this PR. 

I think it's a good thing if there are more or less regular updates  (if they are useful) of `README` and `To-DO` and all these files from time to time.

I created this PR from the conditional-smc2 branch (which was probably an unwise decision as I'll have to eventually merge changes from there into this PR...). I really hope one has to go through this merge conflict mess from time to time for exercise/training purposes and this is actually a good thing... (if there was a smarter solution, please let me know!)

### CODE REVIEW

**Comment by adamjohansen** [comment]
Thanks, @ilyaZar -- it's worth getting these things sorted out. I think if we can merge #60 first it will be rather easier to see what's going on here so perhaps let's get that done and then come back to this?

**Comment by ilyaZar** [comment]
Yes!

**Comment by eddelbuettel** [comment]
Same problem. 21 files changes, If I just set the toggle here to ignore whitespace it is 11 files.

**Comment by ilyaZar** [comment]
> Same problem. 21 files changed -- but if I just set the toggle here to ignore whitespace it is 11 files.

Done. Can delete
```cpp
// -*- mode: C++; c-indent-level: 4; c-basic-offset: 4; indent-tabs-mode: nil; -*-
```
 once @adamjohansen confirms that this is no longer (no hurry though, it's probably wise to do 60 and only then this PR anyway).

**Comment by ilyaZar** [comment]
Done. Another thing on the constency of header files. Currently it is like this:

`// name_of_the_header_file.h: now comes a descriptive text but it tends to be unformatted.`

Just mentioning this here, so we keep in mind whether we want to have this; this refers to one of the `TODO` bullet points "update and standardize all header files" (which is still probably best addressed in this PR).

Obviously for the future, and not relevant now.

**Comment by eddelbuettel** [comment]
I am sorry but where did we say it was a good idea to now delete these lines from all files when we did not do it for a decade?

**Comment by eddelbuettel** [comment]
Please revert https://github.com/rcppsmc/rcppsmc/pull/61/commits/f285885be45e1f0cfe42e8166fca0e6a83ecb08b.

**Comment by ilyaZar** [comment]
Sorry! I accidently assumed deletion because but you wrote "__*could*__ be removed. " and since Adam said it's an artefact I went on removing all (either all or none) of these lines.

My fault, commit reverted now.

**Comment by eddelbuettel** [comment]
I am very sorry if it mislead you. The suggestion I made was for a sole file you created from scratch and where you copied two lines from other files that could be removed.

**Comment by adamjohansen** [comment]
Sorry for confusing things with a quick reply whilst trying to keep a three-year-old happy, I hadn't read the conversation very carefully. Editorconfig is good for people who use it and is more portable than elisp magic; the other lines do have the advantage that they work with native emacs, and not everyone _does_ use editorconfig so it does have some use.

As @eddelbuettel says, let's try to keep the PRs focussed on one thing, anyway, otherwise life gets too complicated.

**Comment by ilyaZar** [comment]
I see, thank you @adamjohansen and @eddelbuettel (indeed I was copy pasting lines that I did not bother to understand correctly :) ) 

I do realize now that it's very good to have a separate PR for these type of things and do very incremental steps (-> even though it might occur that small changes could be just carried out on the fly, they might be not so "small" and innocuous as they seemed!)

**Comment by eddelbuettel** [comment]
As this is a PR, we can write back to your repo I just took the liberty to (first clone it here, set up ours as a parent repo and pull both) rebase via ` git rebase parentrepo/master` and then `git push --force-with-lease origin `.  You may want to make a quick copy or tarball of your local checkout and then pull from your github remote to the force repo.  

Things looks a little cleaner now as this PR is now down to the two commits it had.  Having concurrent / overlapping work in two distinct branches is not a good idea, though sometimes unavoidable.  Rebasing can help, forced pushes are a bit blunt but cleaner histories are worth it.  Or so I think :)

image

**Review by adamjohansen — COMMENTED** [review]
This all looks good to me, except for some rather trivial questions about dates... I don't really mind what the copyright headers look like or how specific we are, but it would be good to be consistent and if we're going to be precise to make sure the dates are more-or-less right....

**Comment by ilyaZar** [comment]
@eddelbuettel thanks for taking time to explain what happened ! (you guessed correctly the importance of this educational moment... otherwise it would've taken quite some time to comprehend...)

@adamjohansen sorry for the numerous super-trivial mistakes, it was probably late when I did those commits; which suggest to commit only in the mornings with a new fresh look at things...

Will fix that asap, but probably tomorrow to not mess around with `git` in evenings (learning from above) and to not include more trivial mistakes... (learning from above).

**Comment by adamjohansen** [comment]
No need to apologise! Just mentioning a few trivial details I noticed, none of this is especially important....

File: inst/include/moveset.h [hunk 1]
```diff
@@ -4,6 +4,7 @@
 //
 // Copyright (C) 2008 - 2009  Adam Johansen
 // Copyright (C) 2017         Adam Johansen, Dirk Eddelbuettel and Leah South
+// Copyright (C) 2017         Adam Johansen, Dirk Eddelbuettel, Leah South and Ilya Zarubin
```
[the remark this answers is not in the document]
Comment (eddelbuettel) [reply inst/include/moveset.h:7]: Good catch.  That line (and similar ones) needs a simple fix:  2021.

File: inst/include/RcppSMC.h [hunk 2]
```diff
@@ -1,4 +1,5 @@
-
+// Copyright (C) 2021 Adam Johansen, Dirk Eddelbuettel, Leah South, Ilya Zarubin
```
[the remark this answers is not in the document]
Comment (eddelbuettel) [reply inst/include/RcppSMC.h:1]: I am a little bit of stickler of not retrofitting work that wasn't done then so I think something like

```
// Copyright (C) 2008 - 2009  Adam Johansen
// Copyright (C) 2017         Adam Johansen, Dirk Eddelbuettel and Leah South
// Copyright (C) 2021         Adam Johansen, Dirk Eddelbuettel, Leah South and Ilya Zarubin
```

is preferable.  I do sometimes fill in to current year when the is maintained so maybe

```
// Copyright (C) 2008 - 2009  Adam Johansen
// Copyright (C) 2017 - 2021  Adam Johansen, Dirk Eddelbuettel and Leah South
// Copyright (C) 2021             Adam Johansen, Dirk Eddelbuettel, Leah South and Ilya Zarubin
```

[the remark this answers is not in the document]
Comment (adamjohansen) [reply inst/include/RcppSMC.h:1]: No objections from me; I take your point and wouldn't argue with it, I just struggle to find the energy to worry about it most of the time.

**Review by LeahPrice — APPROVED** [review]
This looks good to me and I'm happy for it to be merged when the dates are corrected. Thanks!

**Comment by eddelbuettel** [comment]
To keep it simple and moving forward, I just added that in a commit right now.  Good enough?

**Comment by ilyaZar** [comment]
Thanks @eddelbuettel , for me it's fine.

I am doing some final polishing of my GSoC project website since today is the deadline; hence I did not come thus far as to clean my git branches and do this year thing int the `NEWS.Rd`.

**Review by eddelbuettel — APPROVED** [review]
LGTM

**Pull request merged**